### DISCUSSION TITLE
Ractive & Vite?

### DISCUSSION THREAD

**Discussion by pidgpowell** [ractivejs/ractive#3376] [Q&A]
Hi! Just wondering if anyone has got Ractive working with Vite using the rollup plugin? Am experimenting but getting nowhere yet. Will keep working on it but just curious if there's any examples out there. Thanks!

**Comment by evs-chris** [comment]
I haven't tried vite yet, but there is an edge version of the ractive rollup plugin that may play more nicely if vite supports the full rollup plugin api. I'll see if I can poke at it this weekend and report what I find.

**Reply by pidgpowell** [reply]
well, the Vite config was actually pretty straightforward and works with the Ractive rollup plugin as is:

```js
// vite.config.js
import ractive from 'rollup-plugin-ractive-bin'

export default {
  plugins: [
    ractive()
  ]
}
```

I haven't had time yet to get it working using Ractive components with a `.html` extension so am using it with `.ractive` for now. Vite plugins are a superset of the Rollup ones so a lot will work:

https://vitejs.dev/guide/api-plugin.html#rollup-plugin-compatibility

maybe my initial issue was that importing Ractive components requires the file extension e.g.

`import Nav from 'nav.ractive';`
